cmd/query/main.go

```diff
@@ -174,3 +179,26 @@ func registerStaticHandler(r *mux.Router, logger *zap.Logger, qOpts *app.QueryOp
 		logger.Info("Static handler is not registered")
 	}
 }
+
+func archiveOptions(storageFactory istorage.Factory, logger *zap.Logger) []app.HandlerOption {
+	reader, err := storageFactory.CreateSpanReader()
+	if err == istorage.ErrArchiveStorageNotConfigured || err == istorage.ErrArchiveStorageNotSupported {
+		return nil
```

Comment:
Wouldn't this fail silently when `ErrArchiveStorageNotConfigured` is false; meaning it's configured; and when `ErrArchiveStorageNotSupported` is true?
Shouldn't we return the `ErrArchiveStorageNotSupported` when someone mistakenly configures archive storage on a storage that doesn't support archiving?

Reply:
it's not possible to configure archiving on a storage that doesn't support it. For example, with Cassandra you need to pass `--cassandra-archive.enabled=true`.